File: tests/fixtures/languages/java/Animals.java
package com.example.animals;

import java.util.List;
import java.util.ArrayList;

public interface Animal {
    String speak();
}

public class Dog implements Animal {
    private String name;

    public Dog(String name) {
        this.name = name;
    }

    public String speak() {
        return bark();
    }

    private String bark() {
        return name + " says woof";
    }
}

class Puppy extends Dog {
    public Puppy(String name) {
        super(name);
    }
}

File: tests/fixtures/languages/csharp/Animals.cs
using System;
using System.Collections.Generic;

namespace Animals
{
    public interface IAnimal
    {
        string Speak();
    }

    public class Dog : IAnimal
    {
        public string Name { get; set; }

        public Dog(string name)
        {
            Name = name;
        }

        [Route("api/dogs/{id}")]
        [HttpGet]
        public string Speak()
        {
            return Bark();
        }

        private string Bark()
        {
            return Name + " says woof";
        }
    }

    public class Puppy : Dog
    {
        public Puppy(string name) : base(name) { }
    }
}
